### ISSUE TITLE
Error: Command failed: git --no-pager --git-dir

### ISSUE THREAD

**Issue by mylvghb** [rust-lang/rust-analyzer#12445]
**rust-analyzer version**: v0.3.1075

**rustc version**: rustc 1.61.0 (fe5b13d68 2022-05-18)

**cargo config file**: 
``` TOML
[source.crates-io]
replace-with = 'rsproxy'

[source.rsproxy]
registry = "https://rsproxy.cn/crates.io-index"

[registries.rsproxy]
index = "https://rsproxy.cn/crates.io-index"

[net]
git-fetch-with-cli = true
```

**Problem Description**

A recent occurrence

![image](https://user-images.githubusercontent.com/35945793/171538304-dd7b4782-1b4e-447b-b576-b94e84a55bbb.png)

**Comment by HarkerYX** [comment]
also the same problem, how to fix it ?

**Comment by Veykril** [comment]
rust-analyzer does not show diagnostics in Cargo.tomls, that error comes from the `crates` extension

**Comment by Student-operating** [comment]
i have the same problem. have you found a solution ?

**Comment by xhh0223** [comment]
> rust-analyzer does not show diagnostics in Cargo.tomls, that error comes from the `crates` extension

It's true, but I don't know how to solve it
